### PR TITLE
testing user tx error cases

### PR DESCRIPTION
**Pull request by cendhu**
Signed-off-by: senthil <cendhu@gmail.com>

### CODE REVIEW

File: test/user/user_error_cases_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		}
+
+		_, err := s.SubmitTransaction(t, constants.PostUserTx, &types.UserAdministrationTxEnvelope{
+			Payload:   userTx,
+			Signature: testutils.SignatureFromTx(t, s.AdminSigner(), userTx),
+		})
+		require.EqualError(t, err, "TxValidation: Flag: INVALID_DATABASE_DOES_NOT_EXIST, Reason: the database [db1] present in the db permission list does not exist in the cluster")
+	})
+
+	t.Run("creating an user with incorrect certificate would fail", func(t *testing.T) {
```
Comment (tock-ibm) [inline test/user/user_error_cases_test.go:67]: Lets also try a valid certificate but from the wrong CA?

**Comment by tock-ibm** [comment]
@cendhu A question not directly related to this PR: what happens when a user is onboarded correctly, but then, the CA that issued its certificate is removed from Orion? I think that the user will continue to be able to access Orion; Is that what we want?

File: test/user/user_error_cases_test.go (hunk 1 above)
Comment (cendhu) [reply test/user/user_error_cases_test.go:67]: I have created an issue to handle that separately https://github.com/hyperledger-labs/orion-server/issues/396

**Comment by cendhu** [comment]
> @cendhu A question not directly related to this PR: what happens when a user is onboarded correctly, but then, the CA that issued its certificate is removed from Orion? I think that the user will continue to be able to access Orion; Is that what we want?

Only during the enrollment, Certificate is validated against the CA. For signature verification, we only fetch the user's certificate and do not check against the current CA. 

We can fix this by link each user certificate with the CA certificate. When the root CA changes, we can invalidate respective user's certificate (implicitly). Whenever we do a signature verification, we need to check whether the CA's version associated with this certificate has been changed. If so, throw an error. What do you think?

**Comment by tock-ibm** [comment]
> 
We could probably do this when the CA is changed; otherwise it will be to costly, in the hot path of every request. That is, when we change the CA config (note there could be more than one root/intermediary), revalidate all users and put a flag on each one that is not valid. In addition, when we change CAs, we should make sure we don't implicitly invalidate the current admins, or we will lock the admin out of the DB. There is an issue for all that here: https://github.com/hyperledger-labs/orion-server/issues/154

**Comment by tock-ibm** [comment]
@cendhu LGTM. please rebase.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**